tests/components/cast/test_media_player.py

```diff
@@ -440,6 +440,36 @@ async def test_entity_media_states(hass: HomeAssistantType):
     assert state.state == "unknown"
 
 
+async def test_url_replace(hass: HomeAssistantType):
+    """Test functionality of replacing URL for HTTPS."""
+    info = get_fake_chromecast_info()
+    full_info = attr.evolve(
+        info, model_name="google home", friendly_name="Speaker", uuid=FakeUUID
+    )
+
+    chromecast, entity = await async_setup_media_player_cast(hass, info)
+
+    entity._available = True
+    entity.schedule_update_ha_state()
+    await hass.async_block_till_done()
+
+    state = hass.states.get("media_player.speaker")
+    assert state is not None
+    assert state.name == "Speaker"
+    assert state.state == "unknown"
+    assert entity.unique_id == full_info.uuid
+
+    class FakeImage:
+        url = "http://example.com/test.png"
+
+    media_status = MagicMock(images=[FakeImage()])
+    media_status.player_is_playing = True
+    entity.new_media_status(media_status)
+    await hass.async_block_till_done()
+    state = hass.states.get("media_player.speaker")
+    assert state.media_image_url == "//example.com/test.png"
```

Comment:
@KTibow From the test log you can see that the image is present as attribute `entity_picture`:
```
DEBUG:homeassistant.core:Bus:Handling <Event state_changed[L]: entity_id=media_player.speaker, old_state=<state media_player.speaker=unknown; volume_level=<MagicMock name='mock.status.volume_level' id='139796982564816'>, is_volume_muted=<MagicMock name='mock.status.volume_muted' id='139796982573904'>, app_id=<MagicMock name='mock.app_id' id='139797001697232'>, app_name=<MagicMock name='mock.app_display_name' id='139796982098000'>, entity_picture_local=None, friendly_name=Speaker, supported_features=21389 @ 2020-08-06T06:57:01.060321+00:00>, new_state=<state media_player.speaker=playing; volume_level=<MagicMock name='mock.status.volume_level' id='139796982564816'>, is_volume_muted=<MagicMock name='mock.status.volume_muted' id='139796982573904'>, media_content_id=<MagicMock name='mock.content_id' id='139796982310864'>, media_content_type=tvshow, media_duration=<MagicMock name='mock.duration' id='139796981878992'>, media_position=<MagicMock name='mock.current_time' id='139796981892560'>, media_position_updated_at=2020-08-06T06:57:01.062251+00:00, media_title=<MagicMock name='mock.title' id='139796981835728'>, media_artist=<MagicMock name='mock.artist' id='139796981927568'>, media_album_name=<MagicMock name='mock.album_name' id='139796981953296'>, media_album_artist=<MagicMock name='mock.album_artist' id='139796981966736'>, media_track=<MagicMock name='mock.track' id='139796981980304'>, media_series_title=<MagicMock name='mock.series_title' id='139796981927504'>, media_season=<MagicMock name='mock.season' id='139796982027472'>, media_episode=<MagicMock name='mock.episode' id='139796982040912'>, app_id=<MagicMock name='mock.app_id' id='139797001697232'>, app_name=<MagicMock name='mock.app_display_name' id='139796982098000'>, entity_picture_local=/api/media_player_proxy/media_player.speaker?token=c5d12e292941dd862c608b021d62b952c6068b6147c832ada9150a9e6e383b52&cache=376dbe4c67d1f253, friendly_name=Speaker, entity_picture=//example.com/test.png, supported_features=21439 @ 2020-08-06T06:57:01.070021+00:00>>
```

So you need to change this line to:
assert state.attributes.get("entity_picture") == "//example.com/test.png"